### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

to fix security vulnerability and get modren.

## How was this change tested?

unit tests

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
+Lint/MixedRegexpCaptureTypes:
+  Enabled: false
+
 Metrics/BlockLength:
   Exclude:
     - spec/**/*
 
 Metrics/MethodLength:
   Max: 15
-  ExcludedMethods:
```
Comment (jmartin-sul) [inline .rubocop.yml:18]: neat, i didn't even know you could exclude specific methods in the config

Comment (ndushay) [reply .rubocop.yml:18]: surprise!

File: spec/parse_date/int_from_string_spec.rb [hunk 2]
```diff
@@ -698,7 +698,7 @@
       end
     end
     it 'true for 0000' do
```
Comment (jmartin-sul) [inline spec/parse_date/int_from_string_spec.rb:700]: title's now out of sync with test, but wouldn't hold up merge since you said this is a prod issue

**Review by jmartin-sul — APPROVED** [review]
one nitpicky thing about a test title, but otherwise lgtm

**Pull request merged**